src/Container.js

```diff
@@ -66,6 +66,19 @@ export default class Container {
         this._registrations[name] = registration;
         return new RegistrationModifier(registration, this._instanceCache, this._registrationGroups);
     }
+    registerWithResolver(name, resolver, isExternallyOwned = true) {
```

Comment:
I think this is potentially dangerous, as isExternallyOwned is true by default, which is normally the minority of cases with containers.

We need to be careful about naming, because there is already the concept of a resolver in micro-di and this is really an `externallyOwnedFactory`